plugin/trino-postgresql/src/test/java/io/trino/plugin/postgresql/TestingPostgreSqlServer.java

```diff
@@ -62,6 +75,34 @@ private static void execute(String url, Properties properties, String sql)
         }
     }
 
+    protected List<RemoteDatabaseEvent> getRemoteDatabaseEvents()
+    {
+        List<String> logs = getLogs();
+        Iterator<String> logsIterator = logs.iterator();
+        ImmutableList.Builder<RemoteDatabaseEvent> events = ImmutableList.builder();
+        while (logsIterator.hasNext()) {
+            String logLine = logsIterator.next();
+            if (logLine.startsWith("LOG:  execute <unnamed>: ")) {
+                events.add(new RemoteDatabaseEvent(logLine.substring(25), RUNNING));
+            }
+            if (logLine.equals("ERROR:  canceling statement due to user request")) {
+                String cancelledStatementLogLine = logsIterator.next();
```

Comment:
what if there is no next line yet?

Reply:
We will get `NoSuchElementException` :)

There are two options: 
 - ignore it, then I will wonder why query was not cancelled
 - not ignore it and fail (already implemented)
 
 What do you suggest?

Reply:
> not ignore it and fail (already implemented)

this may be the right choice module possible race condition when the next line does not exist _yet_.

however, at least indicate it's intentional choice and not a sloppy coding: `// next line must be present`